internal/bminventory/inventory.go

```diff
@@ -1523,31 +1523,49 @@ func (b *bareMetalInventory) DownloadClusterFiles(ctx context.Context, params in
 }
 
 func (b *bareMetalInventory) DownloadClusterKubeconfig(ctx context.Context, params installer.DownloadClusterKubeconfigParams) middleware.Responder {
+	if apiErr := b.checkFileForDownload(ctx, params.ClusterID.String(), kubeconfig); apiErr != nil {
+		return apiErr
+	}
+
+	respBody, contentLength, err := b.s3Client.Download(ctx, fmt.Sprintf("%s/%s", params.ClusterID, kubeconfig))
+	if err != nil {
+		return installer.NewDownloadClusterKubeconfigConflict().
+			WithPayload(common.GenerateError(http.StatusConflict, errors.Wrap(err, "failed to download kubeconfig")))
+	}
+	return filemiddleware.NewResponder(installer.NewDownloadClusterKubeconfigOK().WithPayload(respBody), kubeconfig, contentLength)
+}
+
+func (b *bareMetalInventory) checkFileForDownload(ctx context.Context, clusterID, fileName string) *common.ApiErrorResponse {
 	log := logutil.FromContext(ctx, b.log)
 	var cluster common.Cluster
-	log.Infof("Download cluster kubeconfig for cluster %s", params.ClusterID)
+	log.Infof("Checking cluster cluster file for download: %s for cluster %s", fileName, clusterID)
 
-	if err := b.db.First(&cluster, "id = ?", params.ClusterID).Error; err != nil {
-		log.WithError(err).Errorf("failed to find cluster %s", params.ClusterID)
+	if !funk.Contains(clusterFileNames, fileName) {
```

Comment:
why clusterFileNames is not taken from swagger?

Reply:
I don't know, but that's how it is...